### DISCUSSION TITLE
moving vms to different node don't update the topology.kubernetes.io/zone

### DISCUSSION THREAD

**Discussion by strelok899** [sergelogvinov/proxmox-cloud-controller-manager#126] [Q&A]
### Description
did live migration of vms to different proxmox node and the topology.kubernetes.io/zone stayed the same
### Environment
kubernetes 1.28
latest plugin
ubuntu 22.04

- Plugin version:
- Kubernetes version: [`kubectl version --short`]
- Node describe: [`kubectl describe node <node>`]
- OS version [`cat /etc/os-release`]

**Comment by sergelogvinov** [comment]
Hello, CCM cannot change region/zone/instance-type/provider-id after creation.
You can delete node resource and restart kubelet, after that kubernetes init the node again with new values.

So, it is not kubernetes way to migrate nodes between zone/region. Better to create new instance and drain/migrate the pods.

**Comment by morsik** [comment]
For future reference, this is easier and faster way to trigger re-labeling ;)

```console
# kubectl taint node $(kubectl get nodes -o name) node.cloudprovider.kubernetes.io/uninitialized:NoSchedule --overwrite
```

I just created fresh cluster as a clone from template VM, so they naturally spawned on 1st Proxmox Node. Recreating them just because of that wasn't worth the time, so triggering relabeling is a lot better and faster after migrating them.
